pkg/exchange/batch/batch.go

```diff
@@ -106,8 +112,23 @@ func (e KLineBatchQuery) Query(ctx context.Context, symbol string, interval type
 					return
 				}
 
-				c <- kline
-				startTime = kline.EndTime.Add(time.Millisecond)
+				workingKLines = append(workingKLines, kline)
+
+				if len(workingKLines) == 200 {
+					c <- workingKLines
+					workingKLines = nil
+				}
+
+				//The issue is in FTX, prev endtime = next start time , so if add 1 ms , it would query forever.
+				startTime = kline.EndTime // .Add(time.Millisecond)
+				tryQueryKlineTimes = 0
+			}
+
+			c <- workingKLines
```

Comment:
rename workingKLines to `batchKLines` might make it easier to understand its purpose.

Reply:
ok